commit fbba8d988c94
Author: Johann Hofmann <jhofmann@mozilla.com>
Date:   Tue May 15 22:02:33 2018 +0200

    Bug 1450448 - Correctly handle non-initialized appCache in site data preferences. r=Gijs, a=RyanVM
    
    MozReview-Commit-ID: IhQGAWNQtR0
    
    --HG--
    extra : rebase_source : c1e902883a977c54ccafd4a1c3042d0a17660996
    extra : source : 9f41afd3012718d839033663196a746a47065007
---
 browser/modules/SiteDataManager.jsm | 14 +++++++++++++-
 1 file changed, 13 insertions(+), 1 deletion(-)

diff --git browser/components/preferences/SiteDataManager.jsm browser/components/preferences/SiteDataManager.jsm
index 118a40ce38fc8..9e08e00c38c5b 100644
--- browser/components/preferences/SiteDataManager.jsm
+++ browser/components/preferences/SiteDataManager.jsm
@@ -192,7 +192,19 @@ var SiteDataManager = {
   },
 
   _updateAppCache() {
-    let groups = this._appCache.getGroups();
+    let groups;
+    try {
+      groups = this._appCache.getGroups();
+    } catch (e) {
+      // NS_ERROR_NOT_AVAILABLE means that appCache is not initialized,
+      // which probably means the user has disabled it. Otherwise, log an
+      // error. Either way, there's nothing we can do here.
+      if (e.result != Cr.NS_ERROR_NOT_AVAILABLE) {
+        Cu.reportError(e);
+      }
+      return;
+    }
+
     for (let group of groups) {
       let cache = this._appCache.getActiveCache(group);
       if (cache.usage <= 0) {
